modules/tests/shared/src/test/scala/io/circe/DecoderSuite.scala

```diff
@@ -561,13 +561,27 @@ class DecoderSuite extends CirceSuite with LargeNumberDecoderTests with TableDri
   it should "fail normally when a field is missing and an optional field is present" in {
     val json = Json.obj("a" -> "1".asJson)
 
-    assert(statefulOpt.decodeJson(json).swap.exists(_.message === "Attempt to decode value on failed cursor"))
+    assert(statefulOpt.decodeJson(json).swap.exists(_.message === "The field b is missing"))
   }
 
   it should "fail normally when a field is missing and an optional field is missing" in {
     val json = Json.obj()
 
-    assert(statefulOpt.decodeJson(json).swap.exists(_.message === "Attempt to decode value on failed cursor"))
+    assert(statefulOpt.decodeJson(json).swap.exists(_.message === "The field b is missing"))
+  }
+
+  it should "fail when a nested field is missing" in {
+    case class Foo(x: Int, bar: Bar)
+    case class Bar(nested: String)
+    val decoder: Decoder[Foo] = new Decoder[Foo] {
+      override def apply(c: HCursor): Decoder.Result[Foo] =
+        for {
+          x <- c.downField("x").as[Int]
+          a <- c.downField("bar").downField("nested").as[String]
+        } yield Foo(x, Bar(a))
+    }
+    val Right(fooJson) = parse("""{"x":42, "bar": {}}""")
+    assert(decoder.decodeJson(fooJson).swap.exists(_.message === "The field bar.nested is missing"))
```

Comment:
I think it would read better if we surrounded the path with `'`s. Also, any reason the path doesn't start with a `.`?